### PR TITLE
Remove User class

### PR DESCRIPTION
**Pull request by timEulitz**
Since only the fundraising-backend repository actually needs a user
object right now, we are moving the class there as a first step to make
the FundraisingStore a bit more lean

When reviewing, maybe double-check to see if I forgot to delete something.

T173364

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -59,6 +59,10 @@ Be aware that the concepts of composer as a script runner and containerization m
 
 ## Release notes
 
+### Version 9.0.0 (2018-08-16)
+
+* Remove `User` entity and move it to `fundraising-backend` repository
```
Comment (gbirke) [inline README.md]: Nice! This removes the necessity to track the other user changes.

**Comment by JeroenDeDauw** [comment]
Really cool!

> only the fundraising-backend repository actually needs a user object **right now**

If at some point another application needs user stuff, it very likely will be different user stuff and should thus not be bound to the FoC user stuff.

**Comment by timEulitz** [comment]
@JeroenDeDauw Hey Jeroen, thanks for your review, unfortunately the code changes in the Factory were WIP and I forgot to mark it as such. I still need to change it for some tests, but no big deal. I am a bit busy with moving so I won't get to change this until Sunday so it's alright to just leave this as it is for now and I will fix it myself.

**Comment by timEulitz** [comment]
@gbirke @tzhelyazkova Please review my latest changes to src/Store/Factory.php which were needed to allow other repositories to specify directories with Doctrine entities in them. After merging this, I can create a release and update the Composer requirements of the backend repository.

**Pull request merged**